[Tan Kel Zin] iP

MechFroG88 opened the pull request:
# Tsohg

**Tsohg** will remind your tasks and events like a ~ghost~. 👻 👻

>uwuwuwuwuwuwu

- Free to use
- Easy to learn
- Fast installation

## Installation

1. Install it from [here](https://github.com/MechFroG88/ip/releases/tag/v0.1)
2. Run it with `java -jar duke.jar`

## Features

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

## Main Function

```java
public static void main(String[] args) {
    new Tsohg("data.txt").run();
}
```

Commit:
Add Gradle support

MechFroG88 committed:
Level-1

MechFroG88 committed:
Level-2

MechFroG88 committed:
Level-3

MechFroG88 committed:
Fix Status Icon

MechFroG88 committed:
Level-4

MechFroG88 committed:
Resolve Merge Conflict

MechFroG88 committed:
Fix Status Icon

MechFroG88 committed:
Fix Spacing Issue

MechFroG88 committed:
A-TextUiTesting

MechFroG88 committed:
Level-5

MechFroG88 committed:
Level-6

MechFroG88 committed:
Level-7

MechFroG88 committed:
Level-8

MechFroG88 committed:
Merge pull request #1 from MechFroG88/branch-Level-7

Level-7

MechFroG88 committed:
Merge branch 'master' into branch-Level-8

MechFroG88 committed:
Merge pull request #2 from MechFroG88/branch-Level-8

Level-8

MechFroG88 committed:
Fix Minor Bug

MechFroG88 committed:
Update A-MoreOOP

MechFroG88 committed:
Update A-Packages

MechFroG88 committed:
Update A-JUnit

MechFroG88 committed:
Add A-JavaDoc

MechFroG88 committed:
Add Level-9

MechFroG88 committed:
Merge pull request #3 from MechFroG88/branch-A-JavaDoc

Add A-JavaDoc

MechFroG88 committed:
Add A-CodingStandard

MechFroG88 committed:
Merge branch 'master' into branch-A-CodingStandard

MechFroG88 committed:
Merge pull request #4 from MechFroG88/branch-A-CodingStandard

Add A-CodingStandard

MechFroG88 committed:
Merge branch 'master' into branch-Level-9

MechFroG88 committed:
Merge pull request #5 from MechFroG88/branch-Level-9

Add Level-9

MechFroG88 committed:
Change Duke to Tsohg

MechFroG88 committed:
Merge branch 'master' of github.com:MechFroG88/ip

MechFroG88 committed:
Change Duke to Tsohg

xMashedxTomatox commented on src/main/java/tsohg/Task.java:
```diff
@@ -0,0 +1,58 @@
+    /**
+     * Marks the task as done.
+     */
+
```
Excess line after javadoc.

xMashedxTomatox commented on src/main/java/tsohg/TaskList.java:
```diff
@@ -0,0 +1,181 @@
+
+    private Storage storage;
+
+    /**
+     * Constructor of the class.
+     * @param storage Storage object to store the list.
+     * @throws TsohgException If something wrong happens.
+     */
```
Can be more specific in the description of the class and error throwing, rather than simply constructor of the class/something wrong.

xMashedxTomatox commented on src/main/java/tsohg/Ui.java:
```diff
@@ -0,0 +1,167 @@
+    private final static Scanner IN = new Scanner(System.in);
+    private final static PrintStream OUT = System.out;
+    private TaskList tasks;
+
```
Missing javadoc for constructor.

xMashedxTomatox commented on src/test/java/StorageStub.java:
```diff
@@ -0,0 +1,30 @@
+import tsohg.Storage;
+
+import java.util.Scanner;
+
```
Missing class javadoc.

xMashedxTomatox reviewed (commented):
Really clean overall, LGTM! Aside from some really small issues, this looks like a really good implementation of the ip.

MechFroG88 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

MechFroG88 committed:
Merge pull request #6 from MechFroG88/add-gradle-support

Add Gradle support

MechFroG88 committed:
Merge branch 'master' of github.com:MechFroG88/ip

MechFroG88 committed:
Add Checkstyle

MechFroG88 committed:
Merge pull request #7 from MechFroG88/add-checkstyle

Add Checkstyle

MechFroG88 committed:
Add GUI

MechFroG88 committed:
Merge pull request #8 from MechFroG88/branch-level-10

Add GUI

MechFroG88 committed:
Add Assertions and Exceptions

The current behavior is that Tsohg will not provide any response when index is out of range for certain commands.

This has to be changed for better user experience.

Exception and more error messages have been added to handle the scenario.

Assertions are also added to ensure the program works well.

MechFroG88 committed:
Merge pull request #9 from MechFroG88/branch-A-Assertions

Add Assertions and Exceptions

MechFroG88 committed:
Improve code quality

MechFroG88 committed:
Improve code quality

MechFroG88 committed:
Merge pull request #10 from MechFroG88/branch-A-CodeQuality

Improve Code Quality

MechFroG88 committed:
Add CI

MechFroG88 committed:
Change folder name

MechFroG88 committed:
Change folder directory

MechFroG88 committed:
Create gradle.yml

MechFroG88 committed:
Merge pull request #11 from MechFroG88/branch-A-CI

Add CI

MechFroG88 committed:
Fix Styles

MechFroG88 committed:
Merge branch 'master' of github.com:MechFroG88/ip

MechFroG88 committed:
Add Priority

MechFroG88 committed:
Merge pull request #12 from MechFroG88/branch-c-priority

Add Priority

MechFroG88 committed:
Improve GUI

MechFroG88 committed:
Set theme jekyll-theme-slate

MechFroG88 committed:
Set theme jekyll-theme-minimal

MechFroG88 committed:
Add User Guide

MechFroG88 committed:
Merge branch 'master' of github.com:MechFroG88/ip

MechFroG88 committed:
Fix styling issues
